### DISCUSSION TITLE
How to extract the consensus sequence in linear genome view?

### DISCUSSION THREAD

**Discussion by gokucoluradu** [GMOD/jbrowse-components#3768] [Q&A] [answered]
Dear all,
I am running JBrowse 2 2.4.2 to display alignments of a bam file against a fasta reference file. This works like charm.
However, I need to extract the consensus sequence of certain positions / the entire genome.

Selecting the region and clicking "get sequence" would only give me the sequence of the reference instead of the consensus.

Is there a way to extract the consensus sequence?

Thank you!

**Comment by cmdcolin** [comment, marked as answer]
we don't currently have any feature to get the consensus sequence from reads. May need to use e.g. http://www.htslib.org/doc/samtools-consensus.html 

I think IGV may also have a feature like this

we could consider ways that it could be added to JBrowse potentially too

**Reply by gokucoluradu** [reply]
Thanks, it would be great if you could include this feature

**Comment by rbuels** [comment]
Sounds like igv.js doesn't support it: https://github.com/igvteam/igv.js/issues/807

**Reply by rbuels** [reply]
I'm wondering if desktop IGV might do it, it's a little hard to tell from the docs and I don't have it on my machine right now.

**Reply by cmdcolin** [reply]
IGV desktop does do it
